### DISCUSSION TITLE
node_options error

### DISCUSSION THREAD

**Discussion by aweakley** [community/community#50667] [Codespaces] [answered]
### Select Topic Area

Question

### Body

In the last few days I've started getting this error in several of my Codespaces across different repositories:

```
node: undefined is not supported in NODE_OPTIONS
```

It looks like NODE_OPTIONS is being set to `undefined`, but previously I think it wasn't being set at all (or maybe it was set to an acceptable value).

```
# env | grep NODE_OPTIONS
NODE_OPTIONS=undefined
```

Unsetting that variable gets rid of my error, so I'm looking for a place in some configuration file to do that. My Dockerfile contains node commands that run successfully during the build, but which generate the error if I run them once the Codespace is running, so I guess the setting is overridden after the build is complete. 

I've tried overriding NODE_OPTIONS in Codespaces secrets, and in `containerEnv` and in `remoteEnv` inside my `devcontainer.json`, but it still gets set to `undefined` during start up somehow.  

Is there a configuration that's read later in the start up process where I could (un)set that value?

**Comment by osortega** [comment, marked as answer]
Hi @aweakley !
`NODE_OPTIONS` seems to be an env variable set by node.
I'm thinking of two possible options to fix this:
1.  Change the node version on your custom image to an older node version that doesn't export the variable.
2. Create a `postCreateCommand`/`postStartCommand`/`postAttachCommand` that unsets the env variable or sets it to a value that doesn't break your scenario.

Hope this helps!

**Reply by aweakley** [reply]
Perfect! Thank you, this works well:

```
"postAttachCommand": "export NODE_OPTIONS="
```
